src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -281,8 +299,12 @@ def values(self) -> ArrayLike:
         >>> idx.values
         array([1, 2, 3])
         """
-        # TODO: SNOW-1458117 implement values
-        return self.to_pandas().values
+        return (
+            self._query_compiler.reset_index()
+            .drop(columns=self._query_compiler.columns)
```

Comment:
it seems a lot of API is trying to do drop and reset_index here, might be useful to extract a common utility here like _convert_index_column_to_data_column, and used for all functions

Reply:
Ok makes sense, this will be a pretty small method though